src/libPMacc/include/eventSystem/tasks/TaskSetValue.hpp

```diff
@@ -174,30 +225,42 @@ class TaskSetValue<T_ValueType, T_dim, true> : public TaskSetValueBase<T_ValueTy
 
     virtual void init()
     {
-        size_t current_size = this->destination->getCurrentSize();
-        const DataSpace<dim> area_size(this->destination->getCurrentDataSpace(current_size));
+        size_t current_size = this->destination->getCurrentSize( );
+        const DataSpace< dim > area_size( this->destination->getCurrentDataSpace( current_size ) );
 
-        if(area_size.productOfComponents() != 0)
+        if( area_size.productOfComponents() != 0 )
         {
             auto gridSize = area_size;
 
-            /* line wise thread blocks*/
-            gridSize.x() = ceil(double(gridSize.x()) / 256.);
+            // size of a chunk(in elements)
+            constexpr int xChunkSize = 256;
```

Comment:
This value is not a CUDA magic number, it is the chunk size for the data without relation to CUDA.
The relation to CUDA is created by `traits::GetNumWorkers<xChunkSize>::value`